src/components/BookInfo/BookInfo.tsx

```diff
@@ -1,50 +1,47 @@
-import { Link, useParams } from 'react-router-dom'
-import { useGoogleBooksApi } from '../hooks/useGoogleApi'
+import { Link } from 'react-router-dom'
+import { useGetBookFromUrl } from '../../hooks/useGetBookFromUrl'
 import './BookInfo.css'
 const BookInfo = () => {
-  const param = useParams()
-
-  const [{ info, isLoading, isError }] = useGoogleBooksApi(`${param.id}`)
-
+  const { book, loading, error } = useGetBookFromUrl()
   return (
     <>
       <Link to={'/bookTask'}>Home</Link>
 
-      <h2>The Details about {info?.volumeInfo.title}</h2>
-      {isLoading && <div className="loader"></div>}
-      {isError ? (
+      <h2>The Details about {book?.title}</h2>
+      {loading && <div className="loader"></div>}
```

Comment:
During `loading` state you are rendering the loader and the empty book details UI at the same time

During loading, only the loader should be rendered